### PR TITLE
refactor: the recent transaction list

### PR DESCRIPTION
**Pull request by guanbinrui**
image

closes #

### CODE REVIEW

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    computedPayload?: EthereumRpcComputed | null,
+) {
+    if (!computedPayload) return receipt?.transactionHash
+    const type = computedPayload.type
+    switch (type) {
+        case EthereumRpcType.SEND_ETHER:
+            return `Send ${formatBalance(
+                (computedPayload._tx.value as string | undefined) ?? '0',
+                nativeTokenDetailed?.decimals,
+            )} ${tokenDetailed?.symbol} to ${formatEthereumAddress(computedPayload._tx.to ?? '', 4)}`
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx:27]: plz extract template variable as named variable

Comment (septs) [reply packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx:27]:
```suggestion
            return `Send ${balance} ${tokenDetailed?.symbol} to ${address}`
```

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        case EthereumRpcType.CONTRACT_INTERACTION:
+            switch (computedPayload.name) {
+                case 'approve':
+                    return `Approve ${formatBalance(computedPayload.parameters?.value, tokenDetailed?.decimals ?? 0)} ${
+                        tokenDetailed?.symbol
+                    } to ${formatEthereumAddress(computedPayload.parameters?.spender ?? '', 4)}`
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx:33]: plz extract template variable as named variable

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                case 'transfer':
+                case 'transferFrom':
+                    return `Transfer ${formatBalance(
+                        computedPayload.parameters?.value,
+                        tokenDetailed?.decimals ?? 0,
+                    )} ${tokenDetailed?.symbol} to ${formatEthereumAddress(computedPayload.parameters?.to ?? '', 4)}`
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/TransactionDescription.tsx:39]: plz extract template variable as named variable

File: packages/maskbook/src/plugins/Wallet/services/addressBook.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { WalletMessages } from '@masknet/plugin-wallet'
+import { ChainId, formatEthereumAddress, isSameAddress } from '@masknet/web3-shared'
+import { uniqBy } from 'lodash'
+import { createPluginDatabase } from '../../../database/Plugin/wrap-plugin-database'
+import { PLUGIN_IDENTIFIER } from '../constants'
+import { currentChainIdSettings } from '../settings'
+
+const MAX_ADDRESS_BOOK_SIZ = 20
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/addressBook.ts:8]: SIZE

File: packages/maskbook/src/plugins/Wallet/services/addressBook.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export async function removeAddress(address: string) {
+    const now = new Date()
+    const chunk = await AddressBookDB.get('address-book', currentChainIdSettings.value)
+    if (!chunk) return
+    await AddressBookDB.add({
+        id: currentChainIdSettings.value,
+        type: 'address-book',
+        records: chunk.records.filter((x) => isSameAddress(x.address, address)) ?? [],
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/addressBook.ts:61]: Wrong condition.

File: packages/maskbook/src/plugins/Wallet/services/recentTransactions/index.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        payload?: boolean
+        computedPayload?: boolean
+    } = {},
+): Promise<RecentTransaction[]> {
+    const { receipt = true, payload = true, computedPayload = true } = options
+    const transactions = await database.getRecentTransactions(address)
+
+    const allSettled = await Promise.allSettled(
+        transactions.map(async ({ hash, payload: payload_ }) => {
+            const receipt_ = await getReceipt(hash)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/recentTransactions/index.ts:39]: Remove unnecessary requests.

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/RecentTransactionList.tsx [hunk 7]
```diff
@@ -47,6 +42,11 @@ export function RecentTransactionList(props: RecentTransactionListProps) {
     const chainId = useChainId()
     const { value: transactions, error, retry } = useRecentTransactions()
 
+    console.log('DEBUG: recent transaction')
+    console.log({
+        transactions,
+    })
+
```
Comment (UncleBill) [inline packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/RecentTransactionList.tsx]: Do you forget to remove this debug code?

**Review by UncleBill — DISMISSED** [review]

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/RecentTransactionList.tsx (hunk 7 above)
Comment (guanbinrui) [reply packages/maskbook/src/plugins/Wallet/SNSAdaptor/WalletStatusDialog/RecentTransactionList.tsx]: DONE

**Pull request merged**